src/argument_parsing/search.cpp

```diff
@@ -98,39 +98,34 @@ void run_search(seqan3::argument_parser & parser)
     // ==========================================
     // Process --pattern.
     // ==========================================
-    
+
     if (parser.is_option_set("pattern"))
     {
         if (arguments.pattern_size < arguments.window_size)
             throw seqan3::argument_parser_error{"The minimiser window cannot be bigger than the sliding window."};
     }
     else
-        
-	// Default behaviour: local match
-	//arguments.pattern_size = arguments.window_size * 2;
-    
-	// Default behaviour: semiglobal match 
+
+    // ==========================================
+    // Set default pattern size.
+    // ==========================================
 	if (!arguments.pattern_size)
 	{
-            std::vector<uint64_t> sequence_lengths{};
-            seqan3::sequence_file_input<dna4_traits, seqan3::fields<seqan3::field::seq>> query_in{arguments.query_file};
-            for (auto & [seq] : query_in | seqan3::views::async_input_buffer(16))
-            {
-                sequence_lengths.push_back(std::ranges::size(seq));
-            }
-            std::sort(sequence_lengths.begin(), sequence_lengths.end());
-            arguments.pattern_size = sequence_lengths[sequence_lengths.size()/2];
+        seqan3::sequence_file_input<dna4_traits, seqan3::fields<seqan3::field::seq>> query_in{arguments.query_file};
+        for (auto & [seq] : query_in | std::views::take(1))
+        {
+            arguments.pattern_size = std::ranges::size(seq) / 2;
         }
```

Comment:
So before it took the median of all sequence lengths?

That means if all sequences have the same length L, then pattern_size was L? That means that half of all sequences wouldn't have worked, if they all had unequal length?

Now it will be the first length halved? 

I still don't completely understand the pattern_size.

Reply:
The pattern size defines the minimum length of a local match. Before it was the median and the app defaulted to a semi-global match strategy if the `--pattern_size` wasn't specified.  Since the app is a local prefilter it should default to a local match strategy. One way to do that is defining minimum match length to be half of the length of the first query. Alternatively, pattern length (i.e min length of a local match) could be a required parameter and I could avoid coming up with an arbitrary default.

Reply:
To be honest, I don't know yet. I have no feeling of what pattern_size is :) As you can see in https://github.com/eaasna/sliding-window/issues/38 I'm still figuring that out xD

I think for now it is fine.